packages/toolpad-app/src/apiErrors.ts

```diff
@@ -0,0 +1,21 @@
+import { ValueOf } from './utils/types';
+
+// Do not export this object directly for easier tree-shaking
+const API_ERROR_CODES = {
```

Comment:
import * as apiErrorCodes from './apiErrorCodes';

export type ApiErrorCode = ValueOf<typeof apiErrorCodes>;

Reply:
i didn't know that worked, thanks! will try it out